src/perform_step/dae_perform_step.jl

```diff
@@ -0,0 +1,36 @@
+function initialize!(integrator, cache::DImplicitEulerCache) end
+
+@muladd function perform_step!(integrator, cache::DImplicitEulerCache, repeat_step=false)
+  @unpack t,dt,uprev,u,f,p = integrator
+  @unpack atmp,tmp = cache
+  alg = unwrap_alg(integrator, true)
+
+  # initial guess
+  guess = tmp
+  guess .= zero(eltype(u))
+
+  zero_func! = (out, x) -> begin
```

Comment:
Not sure if it's worth adjusting this if the nlsolver setup is changed in the near future, but this closure could be fixed using the let block trick.

Reply:
yeah I wouldn't worry about this. The goal will be to quickly get this working with the standard Jacobian/W/nlsolver setup.